src/api/procedures/removeAssetStat.ts

```diff
@@ -0,0 +1,144 @@
+  type: StatType.Balance;
+}
+
+export type RemoveAssetStatParams = { ticker: string } & (
+  | RemoveCountStatParams
+  | RemoveBalanceStatParams
+);
+
+export interface Storage {
+  currentStats: BTreeSetStatType;
+}
+
+/**
+ * @hidden
+ */
+export async function prepareRemoveAssetStat(
+  this: Procedure<RemoveAssetStatParams, void, Storage>,
+  args: RemoveAssetStatParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: { statistics },
+      },
+    },
+    storage: { currentStats },
+    context,
+  } = this;
+  const { ticker, type } = args;
+
+  const tickerKey = stringToTickerKey(ticker, context);
+
+  const op =
+    type === StatType.Count
+      ? statisticsOpTypeToStatOpType(StatisticsOpType.Count, context)
+      : statisticsOpTypeToStatOpType(StatisticsOpType.Balance, context);
+
+  const newStat = statisticsOpTypeToStatType(op, context);
+  const removeIndex = currentStats.indexOf(newStat);
+  const statsArr = currentStats.toArray();
+  statsArr.splice(removeIndex, 1);
+
+  this.addTransaction(
+    checkTxType({
+      transaction: statistics.setActiveAssetStats,
+      args: [tickerKey, statsArr as BTreeSetStatType],
+    })
+  );
+}
+
+/**
+ * @hidden
+ */
+export function getAuthorization(
+  this: Procedure<RemoveAssetStatParams, void, Storage>,
+  { ticker }: RemoveAssetStatParams
+): ProcedureAuthorization {
+  const transactions = [TxTags.statistics.SetActiveAssetStats];
+  const asset = new Asset({ ticker }, this.context);
+  return {
+    permissions: {
+      transactions,
+      assets: [asset],
+      portfolios: [],
+    },
+  };
+}
+
+/**
+ * @hidden
+ */
+export async function prepareStorage(
+  this: Procedure<RemoveAssetStatParams, void, Storage>,
+  args: RemoveAssetStatParams
+): Promise<Storage> {
+  const {
+    context,
+    context: {
+      polymeshApi: {
+        query: { statistics },
+      },
+    },
+  } = this;
+  const { ticker, type } = args;
+
+  const tickerKey = stringToTickerKey(ticker, context);
+  const [currentStats, { requirements }] = await Promise.all([
+    statistics.activeAssetStats(tickerKey),
+    statistics.assetTransferCompliances(tickerKey),
+  ]);
+  const missingStat = !currentStats.find(s => {
+    const stat = meshStatToStatisticsOpType(s);
+    const cmpStat = stat === StatisticsOpType.Balance ? StatType.Balance : StatType.Count;
+    return cmpStat === type;
+  });
+
+  if (missingStat) {
+    throw new PolymeshError({
+      code: ErrorCode.UnmetPrerequisite,
+      message: `Stat of type: "${type}" is not enabled for Asset: "${ticker}"`,
+    });
+  }
+
+  requirements.forEach(r => {
+    if (
+      (type === StatType.Count && r.isMaxInvestorCount) ||
+      (type === StatType.Balance && r.isMaxInvestorOwnership)
+    ) {
+      throw new PolymeshError({
+        code: ErrorCode.UnmetPrerequisite,
+        message:
+          'This statistics cannot be removed because a TransferRequirement is currently using it',
+      });
+    }
+  });
```

Comment:
For `ClaimCount` and `ClaimOwnership` compliance types, you need to consider the claimTypes as well. Also for `MaxInvestorCount` and `MaxInvestorOwnership` you need to check if claimType is `null`